plugins/kotlin/gradle/gradle-java/src/org/jetbrains/kotlin/idea/gradleJava/issues/checkers/GradleUndefinedDaemonJvmCriteriaIssueChecker.kt

```diff
@@ -0,0 +1,39 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package org.jetbrains.kotlin.idea.gradleJava.issues.checkers
+
+import com.intellij.build.issue.BuildIssue
+import org.jetbrains.kotlin.idea.gradleJava.issues.quickfix.GradleAddDaemonToolchainCriteriaQuickFix
+import org.jetbrains.plugins.gradle.issue.ConfigurableGradleBuildIssue
+import org.jetbrains.plugins.gradle.issue.GradleIssueChecker
+import org.jetbrains.plugins.gradle.issue.GradleIssueData
+import org.jetbrains.plugins.gradle.service.execution.GradleExecutionErrorHandler.getRootCauseAndLocation
+import org.jetbrains.plugins.gradle.util.GradleBundle
+
+/**
+ * An [GradleIssueChecker] handling Daemon JVM criteria exception when isn't defined for the project
+ */
+class GradleUndefinedDaemonJvmCriteriaIssueChecker : GradleIssueChecker {
+
+    override fun check(issueData: GradleIssueData): BuildIssue? {
+        val rootCause = getRootCauseAndLocation(issueData.error).first
+        if (false) { // TODO: No known exception since right now is an opt-in feature
```

Comment:
Those aren't related and if replaced is easy to forgot about it throwing  `GradleUndefinedDaemonJvmCriteriaBuildIssue` when the `isDaemonJvmCriteriaRequired ` will return `true`. In addition, version of Gradle isn't necessary in this case since if the criteria is a must indirectly means that version supports it